### DISCUSSION TITLE
android ndk path

### DISCUSSION THREAD

**Discussion by nyuntaka** [airsdk/Adobe-Runtime-Support#2694] [Q&A] [answered]
Publishing with Animate deploy will inevitably result in an error.
I tried installing the latest ndk25.2.9519653 in Android studio and changing the environment variables, but ndk21.4.7075529 is always used. How should I specify the path? No matter how much I search, I don't know.
AIR SDK is latest 50.2.2.6.
Animate is the latest 32.0.2.
![air0](https://github.com/airsdk/Adobe-Runtime-Support/assets/14065274/d53b3021-8240-4151-9ead-174720e17952)

**Comment by nyuntaka** [comment]
Since there was the following wording at the end of the error statement, I will resubmit it to the issue instead of here.
"This should not happen under normal circumstances, please file an issue if it does."

**Reply by ajwfrost** [reply]
That's a very strange error, I can see why they say it shouldn't normally happen!

It's trying to strip/process the "libc++.so" files but seems to think they're not valid object files. And actually when you look at them, they're not ..! they're just text files with some linker instructions in them.

The NDK does look very different so there may be some issue with how it's using this rather than the older NDK, but I suspect there's more of an issue that we need to adjust how this works now. We shouldn't really depend on you having a particular NDK version available/configured..

We'll look into this one further.

**Reply by ajwfrost** [reply]
FYI, when I try this here, it's using NDK 21 but I don't get it as an error, it's saying "BUILD SUCCESSFUL" even though we get this output:

```
Unable to strip library ...AndroidStudioProject\app\build\intermediates\merged_native_libs\debug\out\lib\arm64-v8a\libc++.so due to error 1 returned from ...Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip, packaging it as is.
Successfully started process 'command '...Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip''
Successfully started process 'command '...Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip''
Successfully started process 'command '...Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip''
...Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-strip.exe: error: '...AndroidStudioProject\app\build\intermediates\merged_native_libs\debug\out\lib\armeabi-v7a\libc++.so': The file was not recognized as a valid object file
```

Removing libc++.so from the gradle project structure means it then works, the resulting binary also seems to work.. So we can adjust our Gradle-based build generation for this..

As to why it's showing as an error for you but only a warning for me, I'm not quite sure! As a workaround, you could use AIR SDK Manager to configure the "Keep Android Studio Output" flag, and then once you've got the above failure, you can open the "AndroidStudioProject" folder that will be generated alongside your application files and build from there, you can delete the `libc++.so` files under `AndroidStudioProject\app\src\main\jniLibs\arm64-v8a` (and '..armeabi-v7a', '..x86' and '..x86_64') to get rid of this error.

thanks

**Comment by nyuntaka** [comment]
AIR Package Manager?

**Comment by nyuntaka** [comment]
Sorry, I haven't tried your solution because I still don't get it.
Before that, I tried deleting the ndk/21.4.7075529 folder and running.
There are fewer errors, but I still don't understand the solution.
![air5](https://github.com/airsdk/Adobe-Runtime-Support/assets/14065274/7dde0248-89c6-407c-a4a2-24e69d931d4c)

**Reply by ajwfrost** [reply, marked as answer]
Interesting, that's a different error now, and one that we do have a workaround for: can you please try the suggestion in https://github.com/airsdk/Adobe-Runtime-Support/issues/2608#issuecomment-1590720280

These should be added inside your `<android>` block in the app descriptor XML file (but outside of any `manifestAdditions` etc):
```
  <gradleVersion>7.5</gradleVersion> 
  <androidGradlePluginVersion>7.4.0</androidGradlePluginVersion>
```

**Comment by nyuntaka** [comment]
Solved. Thank you very much.
